starter-code/src/viking.js

```diff
@@ -1,11 +1,95 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+
+    this.attack = function () {
+        return this.strength;
+    }   
+    this.receiveDamage = function (damage){
+        this.health -= damage;
+    }
+}
 
 // Viking
-function Viking() {}
+Viking.prototype = Object.create(Soldier.prototype);
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+
+    this.battleCry = function(){ 
+        return "Odin Owns You All!";
+    }
+    this.receiveDamage = function(damage){
+        this.health -= damage;
+        return (this.health>0 ? this.name + " has received " + damage + " points of damage" : this.name + " has died in act of combat");
+    }
+}
 
 // Saxon
-function Saxon() {}
+Saxon.prototype = Object.create(Soldier.prototype);
+function Saxon(health, strength) {
+    Soldier.call(this,health,strength);
+   
+    this.receiveDamage = function(damage){
+        this.health -= damage;
+        return (this.health>0 ? "A Saxon has received " + damage + " points of damage" : "A Saxon has died in combat");
+    }
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+
+    this.addViking = function(Viking){
+        this.vikingArmy.push(Viking);
+    }
+    this.addSaxon = function(Saxon){
+        this.saxonArmy.push(Saxon);
+    }
+    this.vikingAttack = function(){
+        var randomNumS = (Math.floor(Math.random() * this.saxonArmy.length));
+        var hurtSaxon = this.saxonArmy[randomNumS];
+        
+        var randomNumV = (Math.floor(Math.random() * this.vikingArmy.length));
+        var attackingViking = this.vikingArmy[randomNumV];
+        
+        var saxonStrength = hurtSaxon.receiveDamage(attackingViking.strength);
+
+        for (var i=0; i<this.saxonArmy.length; i++) {
+        if(this.saxonArmy[i].health === 0) {
```

Comment:
El problema está aquí, en ves de ser ...`this.saxonArmy[i].health === 0` quedaría así: `this.saxonArmy[i].health <= 0` ya que a veces va a recibir más daño de la vida que ya tiene, entonces puede dar negativo y eso significa que está muerto 💀